bottom/unpack.py

```diff
@@ -224,29 +224,32 @@ def unpack_command(msg: str) -> Tuple[str, Dict[str, Any]]:
         nickmask(prefix, kwargs)
         kwargs["new_nick"] = params[0]
 
-    elif command == 'RPL_NAMREPLY':
+    elif command in ["RPL_NAMREPLY"]:
         kwargs["target"] = params[0]
-        kwargs["channel_type"] = params[1] if len(params) > 3 else None
+        if len(params) > 3:
+            kwargs["channel_type"] = params[1]
+        else:
+            kwargs["channel_type"] = None
```

Comment:
Good question; in the instances where `message` is optional we provide `""` as the default value.  This is because we can't differentiate omission from empty string.  It simplifies the consumer's side as well since they can rely on `kwargs["message"]` always being a string.

In this case we're populating the default with a different type, which makes it less useful.  However we do have structural information that lets us differentiate empty from omission (not enough parameters -> channel_type was not passed).

A couple options:
1. Make the agreement that all possible kwargs will always be present, and missing values are populated with empty string.  **Pros:** no need to check existence in kwargs **Cons:** empty string may have significant meaning; backwards incompatible if users currently check for `"name" in kwargs` instead of `kwargs.get("name")`.
2. Define the existing contract: where missing/omitted are indistinguishable we'll provide empty string, where unambiguously missing we'll provide None.  **Pros:** No breaking change, users can check `if kwargs["name"] is None` for omission **Cons:** kwargs is no longer a `Dict[str, str]` but a `Dict[str, Optional[str]]`.  Additional type checking overhead even when users know a value will be a string.
3. Preserve existing behavior but use empty string where we use None.  **Pros:** simpler contract means easier to consume **Cons:** Conflating lack of value with empty value may break some commands (ie. arg1 is a remote host when arg2 is populated even if empty, but arg1 is message when arg2 is omitted).

I'm in favor of option 2, where we preserve the `None` but also document that contract extensively so we catch less users off guard when they check `kwarg["something"] == "blah"` and get a `TypeError`.

Thoughts?